File: Programming Basics/06-Nested Loops - Exercise/cinemaTickets.js
function cinemaTickets(input) {

    let index = 0;
    let theInput = input[index];
    index++;

    let countStudent = 0;
    let countStandard = 0;
    let countKid = 0;
    

    while(theInput !== "Finish") {
        let movieName = theInput;
        let freespace = Number(input[index]);
        index++;
        let currentInput = input[index];
        index++
        let currentCountTickets = 0;
        let currentFreeSpace = 0;

        while(currentInput !== "End") {
            if(currentInput === "Finish") {
                break;
            } 
            let typeTicket = currentInput;
            switch(typeTicket) {
                case "student":
                    countStudent++;
                    currentCountTickets++;
                    currentFreeSpace++;
                    break;
                case "standard":
                    countStandard++;
                    currentCountTickets++
                    currentFreeSpace++;
                    break;
                case "kid":
                    countKid++;
                    currentCountTickets++;
                    currentFreeSpace++;
                    break;
            }
            if(currentFreeSpace >= freespace) {
                break;
            }
            currentInput = input[index];
            index++
        }
        let occupancyOfTheMovie = (currentCountTickets / freespace) * 100;
        console.log(`${movieName} - ${occupancyOfTheMovie.toFixed(2)}% full.`);
        
        if(currentInput === "Finish") {
            break;
        }
        theInput = input[index];
        index++;    
    }

    let allTickets = countKid + countStandard + countStudent;
    let ticketPercentegeStudents = (countStudent / allTickets) * 100;
    let ticketPercentegeStandard = (countStandard / allTickets) * 100;
    let ticketPercentegeKid = (countKid / allTickets) *100;
    console.log(`Total tickets: ${allTickets}`);
    console.log(`${ticketPercentegeStudents.toFixed(2)}% student tickets.`);
    console.log(`${ticketPercentegeStandard.toFixed(2)}% standard tickets.`);
    console.log(`${ticketPercentegeKid.toFixed(2)}% kids tickets.`);
    
}

cinemaTickets(["Shutter Island",
"9",
"standard",
"standard",
"standard",
"student",
"student",
"student",
"kid",
"kid",
"kid",
"Rush",
"9",
"standard",
"standard",
"standard",
"student",
"student",
"student",
"kid",
"kid",
"kid",
"Deadpool",
"9",
"standard",
"standard",
"standard",
"student",
"student",
"student",
"kid",
"kid",
"kid",
"Finish"])

File: Programming Basics Java/src/NestedLoopsExercise/CinemaTickets.java
package NestedLoopsExercise;

import java.util.Scanner;

public class CinemaTickets {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String movieName = scanner.nextLine();
        int totalStudentTickets = 0;
        int talalStandardTickets = 0;
        int totalKidsTickets = 0;

        while (!movieName.equals("Finish")) {
            int emptySeats = Integer.parseInt(scanner.nextLine());
            double takenSeats = 0;

            String ticket = scanner.nextLine();

            while (!ticket.equals("End")) {
                switch (ticket) {
                    case "student":
                        totalStudentTickets++;
                        break;
                    case "standard":
                        talalStandardTickets++;
                        break;
                    case "kid":
                        totalKidsTickets++;
                        break;
                }

                takenSeats++;

                if (emptySeats == takenSeats) {
                    break;
                }

                ticket = scanner.nextLine();
            }

            System.out.printf("%s - %.2f%% full.\n", movieName, (takenSeats / emptySeats) * 100);

            movieName = scanner.nextLine();
        }

        int totalSoldTickets = totalStudentTickets + talalStandardTickets + totalKidsTickets;

        System.out.printf("Total tickets: %d\n", totalSoldTickets);
        System.out.printf("%.2f%% student tickets.\n", ((double) totalStudentTickets / totalSoldTickets) * 100);
        System.out.printf("%.2f%% standard tickets.\n", ((double) talalStandardTickets / totalSoldTickets) * 100);
        System.out.printf("%.2f%% kids tickets.\n", ((double) totalKidsTickets / totalSoldTickets) * 100);
    }
}
